hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/expunge/JpaResourceExpungeService.java

```
@@ -315,16 +317,22 @@ private void expungeCurrentVersionOfResource(
 
 		deleteAllSearchParams(JpaPid.fromId(resource.getResourceId()));
 
-		if (resource.isHasTags()) {
-			myResourceTagDao.deleteByResourceId(resource.getId());
-		}
+		try {
+			if (resource.isHasTags()) {
+				myResourceTagDao.deleteByResourceId(resource.getId());
+			}
 
-		if (resource.getForcedId() != null) {
-			ForcedId forcedId = resource.getForcedId();
-			myForcedIdDao.deleteByPid(forcedId.getId());
-		}
+			if (resource.getForcedId() != null) {
+				ForcedId forcedId = resource.getForcedId();
+				myForcedIdDao.deleteByPid(forcedId.getId());
+			}
 
-		myResourceTableDao.deleteByPid(resource.getId());
+			myResourceTableDao.deleteByPid(resource.getId());
+		} catch (Exception e) {
```

minor: we normally don't catch all Exceptions.  It is better to identify the actual exception you are trying to remap.

---

Changed to catch DataIntegrityViolationException only.